library/universes.ml

```diff
@@ -702,12 +702,43 @@ let pr_universe_body = function
 
 let pr_universe_opt_subst = Univ.LMap.pr pr_universe_body
 
-exception Found of Level.t
+let compare_constraint_type d d' =
+  match d, d' with
+  | Eq, Eq | Le, Le | Lt, Lt -> 0
+  | Eq, _ -> -1
+  | Le, Eq -> 1
+  | Le, Lt -> -1
+  | Lt, (Eq | Le) -> 1
```

Comment:
Please make it so that the ordering is obvious. I don't know if it is the most efficient one, but it could be as follows:

    | Eq, Eq -> 0
    | Eq, _ -> -1
    | _, Eq -> 1
    | Le, Le -> 0
    | Le, _ -> -1
    | _, Le -> 1
    | Lt, Lt -> 0